Sources/NIOPosix/NonBlockingFileIO.swift

```diff
@@ -500,3 +499,11 @@ public struct NonBlockingFileIO {
     }
 
 }
+
+extension NIOFileHandle {
+    public func withUnsafeSystemFileDescriptor<T>(_ body: (SystemPackage.FileDescriptor) throws -> T) throws -> T {
```

Comment:
@compnerd/@milseman Does `swift-system` have `FileDescriptor` on Windows and all other platforms?

Reply:
It does have `FileDescriptor` on Windows but, now you mention it, we explicitly opted to _not_ include `resize()` on Windows. See https://github.com/apple/swift-system/pull/82.

Reply:
That is easily accomplished with `SetFilePointerEx` and `SetEndOfFile`; you will need `_get_osfhandle` to convert the file descriptor to the appropriate handle.